### PR TITLE
Add yield when process spill input in hash build

### PR DESCRIPTION
**Pull request by xiaoxmeng**
When hash build process spill input, it does this in a tight
loop until process all the inputs from spill file. This PR
optimizes this by adding yield after processing each spill input
batch.

### CODE REVIEW

File: velox/exec/HashBuild.cpp [hunk 1]
```diff
@@ -912,11 +914,19 @@ void HashBuild::setupSpillInput(HashJoinBridge::SpillInput spillInput) {
 void HashBuild::processSpillInput() {
   checkRunning();
 
+  // TODO: integrate with driver yield framework.
```
Comment (mbasmanova) [inline velox/exec/HashBuild.cpp:917]: Are you thinking of doing something similar to TableScan?

```
if (this->driverCtx_->task->shouldStop() != StopReason::kNone or
          (getOutputTimeLimitMs_ != 0 and
           (getCurrentTimeMs() - startTimeMs) >= getOutputTimeLimitMs_)) {
```

Comment (xiaoxmeng) [reply velox/exec/HashBuild.cpp:917]: I think the driver could have save the timestamp that a driver started to run and provide a advisory method for operator to check in some busy loop to see if they need to jump out. We might want to override the default in some special case. WDYT? Thanks!

Comment (mbasmanova) [reply velox/exec/HashBuild.cpp:917]: This would be nice. Thanks.

File: velox/exec/HashBuild.h [hunk 2]
```diff
@@ -42,17 +42,20 @@ class HashBuild final : public Operator {
   enum class State {
     /// The running state.
     kRunning = 1,
+    /// The yield state that voluntarily yield cpu after running too long when
+    /// processing input from spilled file.
+    kYield = 2,
```
Comment (mbasmanova) [inline velox/exec/HashBuild.h:47]: Curious, why do we need to assign an integer value to each enum?

File: velox/exec/tests/HashJoinTest.cpp [hunk 3]
```diff
@@ -561,18 +561,19 @@ class HashJoinBuilder {
...
       }
     }
   }
 
   void runTest(
       const core::PlanNodePtr& planNode,
       bool injectSpill,
-      int32_t maxSpillLevel = -1) {
+      int32_t maxSpillLevel = -1,
+      uint32_t maxDriverYiledTimeMs = 0) {
```
Comment (mbasmanova) [inline velox/exec/tests/HashJoinTest.cpp:576]: typo: maxDriverYiledTimeMs -> maxDriverYieldTimeMs

**Review by mbasmanova — APPROVED** [review]
Thanks.

File: velox/exec/HashBuild.h (hunk 2 above)
Comment (xiaoxmeng) [reply velox/exec/HashBuild.h:47]: Hmm, yeah, it is not necessary but I always explicitly set value to enum in practice.

**Pull request closed without merging**